### PR TITLE
Draft: least squares map analysis

### PR DESCRIPTION
**Pull request by mreineck**
This aims to implement the alternative approach to Healpix map analysis discussed in #676 .
Please feel free to push improvements to documentation and suggested code changes!
Most specifically I may need help dealing with support for multiple maps and polarization.

### CODE REVIEW

**Comment by mreineck** [comment]
The current version should work for single temperature-only maps. Next planned step is to deal with an IQU-tuple.

**Comment by mreineck** [comment]
Polarization and multiple maps should now work ... I hope I'm doing the correct conversions everywhere.

I think this is now mature enough to discuss naming, interface and documentation details etc. @zonca, any suggestions?

**Comment by mreineck** [comment]
Sorry for the many spurious source changes, those were made by `black`.

**Comment by mreineck** [comment]
@zonca I just remembered that you suggested adding this to `anafast` as an alternative solver, but since the interfaces are rather different (`anafast` does not take a tolerance, and it doesn't return information about convergence) I'm not sure that would be very practical.

**Comment by mreineck** [comment]
Any suggestion how to proceed with this?

**Comment by zonca** [comment]
It seems good to merge for me,
Can you rebase on main and make sure unit tests pass?

**Review by zonca — APPROVED** [review]

**Comment by mreineck** [comment]
I have rebased, but it still says "3 commits"; hope that is OK.

**Comment by zonca** [comment]
good that tests pass,
what about calling the function `map2alm_lsq`? "analysis" is not clear what it means to non-experts

**Comment by mreineck** [comment]
I can do this. My concern with `map2alm` is that it is not really clear from the name which operation is meant: it can be the adjoint of `alm2map` or (some approximation to) its inverse. But I'm still using `map2alm` in my own codes for "map analysis" so this will be at least somewhat consistent.

File: healpy/sphtfunc.py [hunk 1]
```diff
@@ -299,6 +299,115 @@ def map2alm(
...
+
+    Returns
+    -------
+    alm : numpy.ndarray(complex)
+        The reconstructed a_lm coefficients
+    rel_res : float
+        The norm of the residual map (i.e. `map-alm2map(alm)`), divided by the
+        norm of `maps`. This is a measure for the fraction of the map signal that
+        could not be modeled by the a_lm
+    iter : int
```
Comment (zonca) [inline healpy/sphtfunc.py:346]: `iter` is a reserved word in Python, what about using `n_iter` instead?

**Pull request merged**